.github/workflows/test.yml

```diff
@@ -0,0 +1,16 @@
+name: Test
+on: [push]
+env: 
+  CARGO_TERM_COLOR: always
+
+jobs:
+  test:
+    name: cargo test --workspace
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Cache
+        uses: ./.github/actions/cache
+      - run: cargo build --workspace --verbose
+      - run: cargo test --workspace --verbose
```

Comment:
Can I convince you to also build the `examples/ffi`? We've had issues with them bitrotting in the past so it'd be great to have them running in CI :)

Reply:
Yeah, we should definitely add that build to the CI. But it seems I need to basically build the end artifact to make cmake happy. 

I have another proposal in progress, which builds the world using `bazel` its not handling the example yet. 
But once it does - it would just be this invocation in the CI for build checking everything:

``` bash
bazel build //...
```

Bazel or no bazel, we'll definitely need to check this in CI at some point. So I can take a note, to return to it, if we decide bazel sucks too much :)